weasyl/test/searchtag/test_edit_searchtag_blacklist.py

```diff
@@ -0,0 +1,125 @@
+"""
+Tests the searchtag.edit_searchtag_blacklist() function's codepaths.
+
+Abbreviations used in this file:
+    stbl = searchtag blacklist
+"""
+from __future__ import absolute_import
+
+import pytest
+
+from libweasyl import staff
+
+from weasyl import searchtag
+from weasyl.error import WeasylError
+from weasyl.test import db_utils
+
+# Test set of tags
+valid_tags = {'test*', '*test', 'te*st', 'test', 'test_too'}
+invalid_tags = {'*', 'a*', '*a', 'a*a*', '*a*a', '*aa*', 'a**a', '}'}
+combined_tags = valid_tags | invalid_tags
+
+
+@pytest.mark.usefixtures('db')
+def test_edit_user_stbl_with_no_prior_entries():
+    """
+    Verify that SQL code to add tags to the STBL works as expected. Additionally tests as a consequence of the test tag set:
+    
+    - That invalid tags are not added to the ``searchtag`` table.
+    - The ``if added:``, non-global codepath adding the map into ``searchmapuserblacklist``
+    """
+    user_id = db_utils.create_user()
+    tags = searchtag.parse_blacklist_tags(", ".join(combined_tags))
+    searchtag.edit_searchtag_blacklist(user_id, tags)
+    resultant_tags = searchtag.get_searchtag_blacklist(user_id)
+    assert len(resultant_tags) == len(valid_tags)
```

Comment:
Make `valid_tags` and `invalid_tags` lists, `combined_tags = valid_tags + invalid_tags`, and

```python
assert resultant_tags == valid_tags
```